### PR TITLE
[ADD] fieldservice_stage_validation

### PR DESCRIPTION
**Pull request by brian10048**
As the stage of a field service document progresses, it may be important for specific fields of the document be completed.

For example, when a field service order enters the 'Complete' stage, the fields for 'Actual End Date' and/or 'Resolution Notes' should be set.

This module allows you to raise a Validation Error if the designated fields are not set when moving to a new stage. Since stages can be used for field service orders, workers, locations, or equipments, this logic is available to be applied to each document type.

@max3903 Let me know how this looks to you.  I'll add some tests to finish it off if everything looks good and update the settings in base fieldservice to allow the option to install it

### CODE REVIEW

File: fieldservice_stage_validation/models/fsm_stage.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        string='Fields to Validate',
+        help='Select fields which must be set on the document in this stage',
+    )
+
+    stage_type_model_id = fields.Many2one(
+        'ir.model',
+        compute=_compute_stage_model,
+        string='Model for Stage',
+        help='Technical field to hold model type',
+    )
```
Comment (max3903) [inline fieldservice_stage_validation/models/fsm_stage.py:31]: @brian10048 Not sure you need an extra field. What about having a computed domain on validate_fields_ids?

File: fieldservice_stage_validation/models/fsm_equipment.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            stage = rec.stage_id
+            field_ids = stage.validate_field_ids
+            field_names = [x.name for x in field_ids]
+            values = rec.read(field_names)
+
+            for name in field_names:
+                if values[0][name] == False:
+                    raise ValidationError(_('Cannot move to stage "%s" ' \
+                                            'until the "%s" field is set.'
+                                            % (stage.name, name)))
```
Comment (max3903) [inline fieldservice_stage_validation/models/fsm_equipment.py:22]: @brian10048 Does it work without the loop and something like:

```
if any(values[0][field_names]) == False:
    raise ValidationError()
```

Comment (brian10048) [reply fieldservice_stage_validation/models/fsm_equipment.py:22]: @max3903 I get an `unhashable type: 'list'` error when trying `values[0][field_names]`

File: fieldservice_stage_validation/models/fsm_stage.py (hunk 1 above)
Comment (brian10048) [reply fieldservice_stage_validation/models/fsm_stage.py:31]: @max3903 I tried a handful of different options, but can't quite get the same result as I did here

**Comment by max3903** [comment]
@brian10048 Any chance to get this ready to merge soon?

**Comment by smangukiya** [comment]
@brian10048 : fieldservice_stage_validation/static/description/index.html is giving an error while installing this module, can you please take care of it. Other than that everything is working as expected.

**Comment by brian10048** [comment]
@max3903 @smangukiya Everything here should be ready for merge pending your reviews

**Review by smangukiya — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]

**Pull request merged**